File: JS/ContagemDeCedulas.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */

const NOTAS = [100, 50, 20, 10, 5, 2, 1];

let valor = parseInt( gets() );
console.log( valor );

let ar = countNotes( valor );
ar.forEach(function(v, i) {
    console.log( v+" nota(s) de R$ "+NOTAS[i]+",00");
});

/**
 * Count the number of cedules to each note.
 */
function countNotes(total) {
    let arResult = new Array();

    NOTAS.forEach(function (nota, indice, array) {
        if ( total >= nota ) {
            let temp = total / nota;
            arResult[ indice ] = Math.floor( temp );
            total = total % nota;
        } else {
            arResult[ indice ] = 0;
        }
    });
    
    return arResult;
}
module.exports = {countNotes}

File: GO/ContagemDeCedulas.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

var NOTAS = []int{100, 50, 20, 10, 5, 2, 1}

/**
 * Count the number of cedules to each note.
 */
func countNotes(total int) []int {
	arResult := []int{0, 0, 0, 0, 0, 0, 0}

	for i, nota := range NOTAS {
		if total >= nota {
			arResult[i] = total / nota

			total = total % nota
		}
	}

	return arResult
}
